R/Prepare_Census_Tables.R

```diff
@@ -6,89 +6,176 @@
-variableList <- raw[raw$rowType == "Variable Name", 2:4]
-universeList <- raw[raw$rowType == "Table Universe", c(2,4)]
 
-## Add a flag for median variables, so we can produce a warning in the app (medians can't be aggregated)
-tableList$medianFlag <- grepl("^MEDIAN", tableList$stub)
+####  Reformat With One Variable per Row  ####
+## Separate...
+tables <- raw[raw$rowType == "Table Name", c(2,4)]
+variables <- raw[raw$rowType == "Variable Name", 2:5]
+universes <- raw[raw$rowType == "Table Universe", c(2,4)]
+
+#rename certain variables
+tables <- dplyr::rename(tables, tableStub = stub)
+variables <- dplyr::rename(variables, variableStub = stub)
+universes <- dplyr::rename(universes, universeStub = stub)
+
+## ...and merge back properly formatted
+variables <- merge(variables, tables, by = "tableID")
+variables <- merge(variables, universes, by = "tableID")
+
+
+
+####  Identify Parent Variables  ####
+## A parent variable is a variable of the population a variable is a subpopulation of.
+## Some variables have only one (e.g. "Total" is a parent variable of "Male" in B01001),
+## while others have more than one (e.g. "Total" and "Male" are both parent variables of "Under 5 Years" in B01001).
+## In these cases the highest-level parent is "Parent 1" and subsequent parents are "Parent 2," "Parent 3," etc.
+variables <- dplyr::group_by(variables, tableID) %>% dplyr::mutate(indentMax = max(indent))
+
+## A variable is a child if it is a subpopulation of a parent variable above it
+## NOTE: A variable can be a child and not be the most indented variable in its table (see B02003, where "White Alone" is not a parent but is less indented than "White; Asian")
+variables$childFlag <- variables$indent >= lag(variables$indent)
+variables$parentFlag <- variables$indent < lead(variables$indent)
+variables$parentFlag[nrow(variables)] <- F
+
+variables$parentIndex <- NA
+variables$parentIndex[variables$parentFlag == T] <- variables$indent[variables$parentFlag == T]
+
+
+
+
+
+
+
+
+
+
 
-## Table stubs to title case
-tableList$stub <- lettercase::str_title_case(tolower(tableList$stub))
 
-## Variable IDs to stubs, so that all variables will show in the dropdown menu (even when there are two whose stubs would otherwise both be "Under 5" or something like that)
-variableList$stubLong <- paste0(variableList$stub, " (", variableList$variableID, ")")
+##############               TO FIX                   #########################
+
+
+
+
+
+
+
+
+
+
+
+## Create new variables for each parent of a certain variable. Parent 0: highest-level parent
+for(level in c(0,1,2,3,4,5)) {
+  
+  varname <- paste0("parent", level)
+  varnameStub <- paste0("parent", level, "Stub")
+  
+  variables[[varname]] <- NA
+  variables[[varname]][1] <- "temp"
+  variables[[varname]][variables$indent == level & variables$parentFlag == T] <- variables$variableID[variables$indent == level & variables$parentFlag == T]
+  variables[[varname]] <- zoo::na.locf(variables[[varname]])
```

Comment:
Might want to consider adding `install.packages("zoo")` so that folks know to install this to successfully run this script.

Reply:
added `library(zoo)` at the top, is this not the same?

Reply:
It does! Thank you.